### PR TITLE
Fix Windows issues

### LINKED ISSUE #6: Not portable; broken on Windows
For Windows, `replace-in-files-cli` can't deal with quotes or white-space for arguments and all CI tests fail (see <https://ci.appveyor.com/project/rivy/js-replace-in-files-cli/builds/27906120>).

Happy to help if wished.

### PR DESCRIPTION
**Pull request by rivy**
> fixes #6 

Several fixes here, separated by commit...

- corrected glob expressions supplied to `globby` (must be normalized to forward slashes, and so failed for windows)
- fixed a permissions issue for `tmpdir()` causing 'globs' test to fail on some machines
- fixed a missing `catch` for internal faults
- fixed 'no globs' test specifically for Windows (removed use of an "impossible" file)

It's based on my [AppVeyor PR](https://github.com/sindresorhus/replace-in-files-cli/pull/7) for testing, but that's easily removed if you want to go pure Travis CI.

I can rework and/or drop any of the commits, by your request.

### CODE REVIEW

**Comment by rivy** [comment]
AppVeyor CI tests are [passing](https://ci.appveyor.com/project/rivy/js-replace-in-files-cli/builds/27909277).

File: api.js [hunk 1]
```diff
@@ -1,7 +1,10 @@
 'use strict';
+const isWinOS = /^win/i.test(process.platform);
```
Comment (sindresorhus) [inline api.js]: This is not the correct way to test Windows.
```suggestion
const isWindows = process.platform === 'win32';
```

**Comment by sindresorhus** [comment]
I think it would be better for the glob library to expose a normalization function, rather than every package having to reimplement the logic: https://github.com/mrmlnc/fast-glob/issues/240

File: cli.js [hunk 2]
```diff
@@ -69,4 +69,7 @@ if (cli.flags.replacement === undefined) {
 		ignoreCase: cli.flags.ignoreCase,
 		glob: cli.flags.glob
 	});
-})();
+})().catch(error => {
+	console.error(error);
+	process.exit(1);
+});
```
Comment (sindresorhus) [inline cli.js]: This is moot. `meow` handles this for us.

**Comment by sindresorhus** [comment]
You can add Window to Travis instead, like done here: https://github.com/sindresorhus/del/blob/master/.travis.yml

**Comment by rivy** [comment]
> I think it would be better for the glob library to expose a normalization function, rather than every package having to reimplement the logic: [mrmlnc/fast-glob#240](https://github.com/mrmlnc/fast-glob/issues/240)

Reasonable, but `globby` is yours and neither it nor `fast-glob` work that way as of yet.
This corrects the issue for the time being until the glob libraries expose the behavior.
You could add a `ToDO` comment to refactor the repair as you suggest in the future.

**Comment by rivy** [comment]
> You can add Window to Travis instead, like done here: https://github.com/sindresorhus/del/blob/master/.travis.yml

As you wish, I'll drop the AppVeyor commit and add a Travis commit in the form you suggested here, likely this evening.

File: api.js (hunk 1 above)
Comment (rivy) [reply api.js]: Changed in *api.js* and *test.js*.

File: cli.js (hunk 2 above)
Comment (rivy) [reply cli.js]: I don't remember the exact issue, but my commit note says that this is needed to fix a failing test ('no globs').

Comment (rivy) [reply cli.js]: It's been a few weeks... I don't see the error now. 
I'll drop the commit.

**Comment by kevincaradant** [comment]
Hi guys

Any news to merge it ? Thanks :)

**Pull request merged**